apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/lib/site-intent/use-site-intent.js

```diff
@@ -2,17 +2,24 @@ import apiFetch from '@wordpress/api-fetch';
 import { useState, useEffect, useCallback } from '@wordpress/element';
 
 const useSiteIntent = () => {
-	const [ siteIntent, setSiteIntent ] = useState( '' );
+	const [ siteIntent, setSiteIntent ] = useState( undefined );
+	const [ siteIntentFetched, setSiteIntentFetched ] = useState( false );
 
 	const fetchSiteIntent = useCallback( () => {
 		apiFetch( { path: '/wpcom/v2/site-intent' } )
-			.then( ( result ) => setSiteIntent( result.site_intent ) )
-			.catch( () => setSiteIntent( '' ) );
+			.then( ( result ) => {
+				setSiteIntent( result.site_intent );
+				setSiteIntentFetched( true );
+			} )
+			.catch( () => {
+				setSiteIntent( undefined );
+				setSiteIntentFetched( true );
+			} );
 	}, [] );
 
 	useEffect( () => {
 		fetchSiteIntent();
 	}, [ fetchSiteIntent ] );
-	return siteIntent;
+	return { siteIntent, siteIntentFetched };
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg)
We can consider returning an array so that we don't need to rename it every existing usage

Reply:
This should be fixed by replacing the hook itself!
https://github.com/Automattic/wp-calypso/pull/73565#discussion_r1118211150